pkg/local_workflows/report_analytics_workflow.go

```diff
@@ -41,25 +42,32 @@ func reportAnalyticsEntrypoint(invocationCtx workflow.InvocationContext, inputDa
 	url := fmt.Sprintf("%s/rest/api/orgs/%s/analytics", config.GetString(configuration.API_URL), config.Get(configuration.ORGANIZATION))
 
 	for i, input := range inputData {
-		logger.Println(fmt.Sprintf("%s: processing element %d", reportAnalyticsWorkflowName, i))
+		logger.Printf(fmt.Sprintf("%s: processing element %d", reportAnalyticsWorkflowName, i))
 		documentLoader := gojsonschema.NewBytesLoader(input.GetPayload().([]byte))
-		result, err := gojsonschema.Validate(scanDoneSchemaLoader, documentLoader)
+		result, validationErr := gojsonschema.Validate(scanDoneSchemaLoader, documentLoader)
 
-		if err != nil {
-			logger.Printf("Error validating input: %v\n", err)
-			break
+		if validationErr != nil {
+			err := fmt.Errorf("error validating input at index %d: %w", i, validationErr)
+			logger.Printf("%v\n", err)
+			return nil, err
 		}
 
 		if !result.Valid() {
-			return nil, fmt.Errorf("Error validating input: %v\n", result.Errors())
+			err := fmt.Errorf("validation failed for input at index %d: %v", i, result.Errors())
+			logger.Printf("%v\n", err)
+			return nil, err
 		}
 
-		err = callEndpoint(invocationCtx, input, url)
-		if err != nil {
-			return nil, fmt.Errorf("Error calling endpoint: %v\n", err)
+		callErr := callEndpoint(invocationCtx, input, url)
+		if callErr != nil {
+			err := fmt.Errorf("error calling endpoint for input at index %d: %w", i, callErr)
+			logger.Printf("%v\n", err)
+			return nil, err
 		}
+
 	}
-	return nil, err
+	logger.Println(reportAnalyticsWorkflowName + " workflow end")
+	return nil, nil
```

Comment:
Since I'm adding more context to the error messages, I've decided to create error variables for each scope. I'm more than happy to revert to overriding the `err` variable if that aligns better with the team's convention :)